### PR TITLE
MUSARK-532

### PR DESCRIPTION
**Pull request by rituvesh**
…and second level

### CODE REVIEW

File: src/components/magasin/panel/index.js [hunk 1]
```diff
@@ -193,7 +194,26 @@ export default class StorageUnitContainer extends Component {
     );
   }
 
+  getNodeTypeItems() {
```
Comment (jarlah) [inline src/components/magasin/panel/index.js]: this implementation does not follow the jira issue description " (...) Dersom bruker velger en type som ikke er lovlig, må det gis melding om dette (tilsvarende som for andre valideringsfeil (...)".

In addition, I am not able to understand with a reasonable attempt what this method actually computes. I cannot understand what and why its doing.

File: src/components/magasin/panel/index.js [hunk 2]
```diff
@@ -193,7 +194,26 @@ export default class StorageUnitContainer extends Component {
...
+    // First level 'Organisation'. Second level 'Building'
+    const nodeType = ['StorageUnit', 'Room', 'Building', 'Organisation'];
```
Comment (jarlah) [inline src/components/magasin/panel/index.js]: nodeTypes

File: src/components/magasin/panel/index.js [hunk 3]
```diff
@@ -193,7 +194,26 @@ export default class StorageUnitContainer extends Component {
...
+    const nodeTypeOrganisation = ['Organisation'];
```
Comment (jarlah) [inline src/components/magasin/panel/index.js]: nodeTypesOrg...

File: src/components/magasin/panel/index.js [hunk 4]
```diff
@@ -193,7 +194,26 @@ export default class StorageUnitContainer extends Component {
...
+    const nodeTypeBuilding = ['Building'];
+    let lv_return;
+    if (this.props.rootNode &&
```
Comment (jarlah) [inline src/components/magasin/panel/index.js]: const { pathNames } = this.props.rootNode || {}
if (pathNames && pathNames.length > .. ..... ) {
   ....
}

File: src/components/magasin/panel/index.js [hunk 5]
```diff
@@ -233,7 +253,7 @@ export default class StorageUnitContainer extends Component {
                                     tooltip={this.props.translate('musit.storageUnits.type.tooltip')}
                                     placeHolder={this.props.translate('musit.storageUnits.type.placeHolder')}
                                     maximumLength={100}
-                                    items={['StorageUnit', 'Room', 'Building', 'Organisation']}
+                                    items={this.getNodeTypeItems()}
```
Comment (jarlah) [inline src/components/magasin/panel/index.js]: getAllowedNodeTypes

**Review by jarlah — CHANGES_REQUESTED** [review]

File: src/components/magasin/panel/index.js (hunk 1 above)
Comment (rituvesh) [reply src/components/magasin/panel/index.js]: NA

File: src/components/magasin/panel/index.js (hunk 2 above)
Comment (rituvesh) [reply src/components/magasin/panel/index.js]: Done.

File: src/components/magasin/panel/index.js (hunk 3 above)
Comment (rituvesh) [reply src/components/magasin/panel/index.js]: Done.

File: src/components/magasin/panel/index.js (hunk 4 above)
Comment (rituvesh) [reply src/components/magasin/panel/index.js]: Done.

File: src/components/magasin/panel/index.js (hunk 5 above)
Comment (rituvesh) [reply src/components/magasin/panel/index.js]: Done.

**Review by jarlah — APPROVED** [review]

**Pull request merged**